src/client/Microsoft.Identity.Client/MsalServiceExceptionFactory.cs

```diff
@@ -127,6 +128,23 @@ private static bool IsThrottled(OAuth2ResponseBase oAuth2Response)
             return ex;
         }
 
+        internal static MsalServiceException FromManagedIdentityResponse(
+            string errorCode,
+            HttpResponse httpResponse)
+        {
+            var managedIdentityResponse = JsonHelper.TryToDeserializeFromJson<ManagedIdentityErrorResponse>(httpResponse?.Body);
+
+            string message = managedIdentityResponse == null ? 
+                "Empty error response received." :
```

Comment:
I don't think a user will have enough info to go if they get this.

Reply:
I think it would be ideal if the exception messages had the MI Source name .

Reply:
I am planning to remove this from here and create the message in the managed identity part of the code as error handling is different for different sources.